hippod/report_generator.py

```diff
@@ -211,7 +217,10 @@ def add_achievement(self, description_path, achievement_path, title, \
                 submitter = achievement_content['submitter']
                 test_date = achievement_content['test-date']
                 categories = categories
-                responsible = attach_content['responsible']
+                if attach_content != None:
+                    responsible = attach_content['responsible']
+                else:
+                    responsible = 'No responsible'
```

Comment:
the whole responsible checking seems to hacky! If the test is uploaded, and has no responsible, why not set at this point anonymous at this point? Means that there will **always** a responsible entry and the entry has a corresponding name in the user db. Solved a lot of things and make the code easier.

I think we are suffering from complexity problems now, too many bugs are triggered the last time, we should try hard to reduce complexity by engineering tricks like this. Reducing branches (if else) in the code by setting a default, always and in the subsequent code no if/else is required.